src/Microsoft.AspNetCore.Mvc.Core/Internal/ResourceInvoker.cs

```diff
@@ -827,7 +950,7 @@ private Task Next(ref State next, ref Scope scope, ref object state, ref bool is
                         Debug.Assert(state != null);
                         Debug.Assert(_resultExecutingContext != null);
 
-                        var filter = (IResultFilter)state;
+                        var filter = (TFilter)state;
```

Comment:
@kichalla \ @rynowak do you think there's value in distinguishing between result filters and always run result filters during logging? With the new logging changes, we'll log them as "Result Filter" which IMO seems fine.

Reply:
I think there could be some value in it - **IF** we can do a good job. The logging that @kichalla added is intelligible and reasonable. I don't want to sacrifice that.

Reply:
Here's what the logs look like:

```
trce: Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker[3]
      Resource Filter: After executing OnResourceExecuting on filter Microsoft.AspNetCore.Mvc.ConsumesAttribute.
dbug: Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker[4]
      Request was short circuited at resource filter 'Microsoft.AspNetCore.Mvc.ConsumesAttribute'.
trce: Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker[2]
      Always Run Result Filter: Before executing OnResultExecuting on filter BasicWebSite.UnprocessableResultFilter.
trce: Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker[3]
      Always Run Result Filter: After executing OnResultExecuting on filter BasicWebSite.UnprocessableResultFilter.
trce: Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker[4]
...
```